vassal-app/src/main/java/VASSAL/build/module/chessclockcontrol/ChessClock.java

```diff
@@ -0,0 +1,826 @@
+import VASSAL.build.BadDataReport;
+import VASSAL.build.Buildable;
+import VASSAL.build.GameModule;
+import VASSAL.build.module.ChessClockControl;
+import VASSAL.build.module.GameComponent;
+import VASSAL.build.module.PlayerRoster;
+import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.command.Command;
+import VASSAL.command.CommandEncoder;
+import VASSAL.command.NullCommand;
+import VASSAL.configure.ColorConfigurer;
+import VASSAL.configure.Configurer;
+import VASSAL.configure.ConfigurerFactory;
+import VASSAL.configure.IconConfigurer;
+import VASSAL.configure.TranslatableStringEnum;
+import VASSAL.i18n.Resources;
+import VASSAL.tools.ErrorDialog;
+import VASSAL.tools.LaunchButton;
+import VASSAL.tools.SequenceEncoder;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * CHESS CLOCK class for VASSAL.
+ *
+ * The ChessClock class itself implements a single timer, which is added as a button to the Module toolbar.
+ * Clicking the button starts and stops the clock, which we attempt to keep (roughly) in sync across multiple
+ * players' computers. A pair (or more, or I suppose only one) of them are then added to {@link ChessClockControl}.
+ *
+ * @author Brian Reynolds and Michael Kiefte
+ *
+ * Class originally created by Michael Kiefte for Twilight Struggle, including handshaking for verification of time totals between online
+ * machines. Made more generically configurable, ChessClockControl added, and integrated into standard VASSAL, by Brian Reynolds.
+ *
+ */
+public class ChessClock extends AbstractConfigurable implements CommandEncoder, GameComponent, ActionListener {
+  private static final int MILLISECONDS_PER_MINUTE = 1000 * 60;
+  private static final int MILLISECONDS_PER_HOUR = MILLISECONDS_PER_MINUTE * 60;
+  private static final int MILLISECONDS_PER_DAY = MILLISECONDS_PER_HOUR * 24;
+  public static final char DELIMITER = '\t';
+  public static final String COMMAND_PREFIX = "CLOCK" + DELIMITER; //$NON-NLS-1$
+
+  public static final String ICON = "icon"; //$NON-NLS-1$
+  public static final String SIDE = "side"; //$NON-NLS-1$
+  public static final String TOOLTIP = "tooltip"; //$NON-NLS-1$
+  public static final String BUTTON_TEXT = "buttonText"; //$NON-NLS-1$
+
+  public static final String TICKING_BACKGROUND_COLOR = "tickingBackgroundColor";
+  public static final String TICKING_FONT_COLOR       = "tickingFontColor";
+  public static final String TOCKING_FONT_COLOR       = "tockingFontColor";
+
+  public static final String GENERIC   = "Player";
+
+  private static Color defaultColor;       // Stores default look-and-feel colors for buttons
+  private static Color defaultFontColor;
+
+  protected Color tickingBackgroundColor = new Color(255, 255, 0);  // Starting (default) colors
+  protected Color tickingFontColor       = new Color(51, 51, 51);
+  protected Color tockingFontColor       = new Color(0, 0, 0);
+
+  protected static Timer timer;    // Millisecond timer (but only actually updated every ~100ms)
```

Comment:
If you have only one timer, should it be non-static and in the `ChessClockControl` instead of static and here?

Reply:
A thing I liked about the static version was that, in the admittedly oddball case where someone had *multiple* ChessClock controllers, they'd all still share a single timer instance.